sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -960,9 +970,19 @@ protected override Task UpdateCheckpointAsync(string partitionId,
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
             Argument.AssertNotNull(partitionId, nameof(partitionId));
-            Argument.AssertAtLeast(startingPosition.SequenceNumber, 0, nameof(startingPosition.SequenceNumber));
+            if (string.IsNullOrEmpty(startingPosition.OffsetString))
+            {
+                if (EventHubProperties?.IsGeoReplicationEnabled ?? false)
+                {
+                    var message = string.Format(CultureInfo.InvariantCulture, Resources.ProcessorAttemptingToWriteCheckpointWithoutOffset);
+                    var updateCheckpointException = new EventHubsException(true, EventHubName, message, EventHubsException.FailureReason.GeneralError);
+                    _ = InvokeOnProcessingErrorAsync(updateCheckpointException, Resources.OperationEventProcessingLoop, CancellationToken.None);
+                }
 
-            Logger.UpdateCheckpointStart(partitionId, Identifier, EventHubName, ConsumerGroup);
+                Argument.AssertAtLeast(startingPosition.SequenceNumber, 0, nameof(startingPosition.SequenceNumber));
```

Comment:
After looking that over - it's correct.   The intent is to say "if we don't have an offset, then make sure we have a sequence number."     If the offset exists (not null or empty) then we don't need to validate sequence number because we don't care about it.

For a while, that was flipped - then the service implementation changed so we're back to using the offset as a primary and falling back to sequence number.

Reply:
Ok yeah that makes sense!